src/main/java/org/springframework/data/couchbase/repository/CouchbaseRepository.java

```diff
@@ -43,4 +43,8 @@
 	@Override
 	List<T> findAllById(Iterable<ID> iterable);
 
+	<S extends T> S replace(S var1);
```

Comment:
Adding technology-specific methods violates somewhat the concept of handling technology-specific aspects inside the repository/template API. I left a comment on DATACOUCH-623. Especially entities annotated with `@Version` shouldn't require the user to call a different method for proper handling, instead, the requirement to issue a different store-specific replace call should be determined by the Spring Data code and not the caller.

Reply:
Thanks Mark.  I put the upsert() vs. replace() logic in the save().